schema-to-pojo-org-json/src/main/java/org/sagebionetworks/schema/adapter/org/json/JSONObjectAdapterImpl.java

```diff
@@ -211,7 +211,11 @@ public JSONObjectAdapter putNull(String key)
 	public JSONObjectAdapter put(String key, double value)
 			throws JSONObjectAdapterException {
 		try {
-			wrapped.put(key, value);
+			if (Double.isNaN(value) || Double.isInfinite(value)) {
+				wrapped.put(key, Double.toString(value));
+			} else {
+				wrapped.put(key, value);
+			}
 			return this;
 		} catch (JSONException e) {

```

Comment:
There are two implementations of the adapter.  The same fix would be needed for both.